tests/integration/common/state.rs

```diff
@@ -85,25 +98,286 @@ fn init_logging() {
         .expect("Failed to configure env_logger");
 }
 
+#[derive(Debug)]
+struct Cairo0Contract {
+    deprecated_compiled_class: DeprecatedCompiledClass,
+    /// Contract address to use when deploying the contract.
+    address: ContractAddress,
+}
+
+#[derive(Debug)]
+struct Cairo1Contract {
+    contract_class: ContractClass,
+    compiled_contract_class: CasmContractClass,
+    /// Contract address to use when deploying the contract.
+    address: ContractAddress,
+}
+
+#[derive(Debug)]
+struct FeeConfig {
+    strk_fee_token_address: ContractAddress,
+    eth_fee_token_address: ContractAddress,
+    initial_balance: u128,
+}
+
+/// Builds the initial state for OS integration tests.
+///
+/// This builder allows to deploy contracts and configure account balances and then generates
+#[derive(Debug)]
+struct StarknetStateBuilder<'a> {
+    /// Cairo 0 contracts (name -> contract class).
+    cairo0_contracts: HashMap<String, Cairo0Contract>,
+    /// Cairo 1 contracts (name -> contract class + compiled contract class).
+    cairo1_contracts: HashMap<String, Cairo1Contract>,
+    /// Config for fees: initial balance, fee token contract addresses, etc.
+    fee_config: Option<FeeConfig>,
+    block_context: &'a BlockContext,
+    /// Contract address generator.
+    address_generator: StdRng,
+}
+
+impl<'a> StarknetStateBuilder<'a> {
+    fn new(block_context: &'a BlockContext) -> Self {
```

Comment:
Any reason not to make these `pub`?

Reply:
No, it's true that there's a lot of private methods so I'll make the ones that need to be `pub`.